pylint/lint.py

```diff
@@ -89,6 +90,23 @@
 MANAGER = astroid.MANAGER
 
 
+def _ast_from_string(data, filepath, modname):
+    cached = MANAGER.astroid_cache.get(modname)
+    if cached and cached.file == filepath:
+        return cached
+
+    from astroid.builder import AstroidBuilder
```

Comment:
We can import the builder at the top btw, there's no circular import to be avoided here.